### PR TITLE
paywall fix

### PR DESCRIPTION
**Pull request by nynevi**
added an alert if the story has one section because paywalls cannot work with only one section. also lists cannot be broken into two parts due to technical limits

### CODE REVIEW

**Comment by nynevi** [comment]
lists are split on backend so I was wrong, I will check the parsing algorithm to make it work with lists

**Comment by nynevi** [comment]
it is almost impossible to correctly parse incorrect html so lists turn out to be wrong.

I matched the element count of frontend and backend so if user sees 1 section only, he cannot enable paywalls. there is a check for this on the backend if used via api

File: src/editor/EditorCtrl.ts [hunk 1]
```diff
@@ -265,6 +266,9 @@ export default class EditorCtrl {
 
     const refreshBodies = (externalHtml?) => {
       fixedBody = this.editorService.getFixedBody(bodyEditor, externalHtml);
+      $scope.hasOnlyOneSection = $(fixedBody).length === 0 || $(fixedBody).length === 1 ?
+        true :
+        false;
```
Comment (adrianbarwicki) [inline src/editor/EditorCtrl.ts]: Is it equivalent to:  `$(fixedBody).length === 0 || $(fixedBody).length === 1` ?

Comment (adrianbarwicki) [reply src/editor/EditorCtrl.ts]: Is it the same as:  `$(fixedBody).length < 2` ?

File: public/templates/layout_write.html [hunk 2]
```diff
@@ -81,9 +81,10 @@ <h4 class="modal-title" id="myModalLabel"><b>Prepare your story for readers</h4>
 
 					<div class="form-group m-0">
 						<form>
-							<div class="checkbox m-0">
+							<div class="checkbox m-0" ng-class="{'disabled': hasOnlyOneSection}">
 								<label>
-									<input type="checkbox" ng-checked="draft.hasPaidSection" ng-click="togglePaidSection()"> <strong>I have a paid section in my story</strong>
+									<input type="checkbox" ng-checked="draft.hasPaidSection" ng-disabled="hasOnlyOneSection" ng-click="togglePaidSection()" ng-class="{'disabled': hasOnlyOneSection}"> <span ng-class="{'bold': !hasOnlyOneSection, 'disabled': hasOnlyOneSection}">I have a paid section in my story</span>
+									<strong ng-if="hasOnlyOneSection" style="color: red;">&nbsp;<i class="fa fa-exclamation-circle" aria-hidden="true"></i> Please split your story into two sections</strong>
```
Comment (adrianbarwicki) [inline public/templates/layout_write.html]: What is the purpose of `&nbsp;`?

**Comment by adrianbarwicki** [comment]
After a merge of another PR, there are merge conflicts.

**Comment by nynevi** [comment]
solved conflicts

**Comment by adrianbarwicki** [comment]
@nynevi could you name the next PRs by the convention <ticketnumber>-shortdesc.

**Comment by adrianbarwicki** [comment]
@nynevi when the story cannot be spit into two and the paywall option is selected, the publish button should be disabled.

**Comment by adrianbarwicki** [comment]
There are conflicts with dev after the linting merge. @nynevi

**Comment by nynevi** [comment]
- fixed merge conflicts
- added bootstrap styling to currency inputs
- added a reset function for draft's paywall settings if the post later had its body changed to one section to automatically reset it

**Comment by nynevi** [comment]
- solved merge conflicts

**Comment by nynevi** [comment]
closing this as it is no longer necessary because the functionality is in v2 now

**Pull request closed without merging**